src/ast/codegen_llvm.cpp

```diff
@@ -753,28 +753,48 @@ void CodegenLLVM::visit(Binop &binop)
   Type &type = binop.left->type.type;
   if (type == Type::string)
   {
-    Value *val;
+    Value *left_string = nullptr;
+    Value *right_string = nullptr;
```

Comment:
I feel like the naming of left and right here is a bit confusing since they don't necessarily match up to positions in the original expression.

It would potentially be simpler to save `bool inverse = binop.op == bpftrace::Parser::token::NE` (but with error checking) right at the start and then call `CreateStrcmp(expr_, string_literal, inverse)` directly in each of your `is_literal` if statements rather than saving the strings to new values.